mesonbuild/interpreter.py

```diff
@@ -3983,6 +3983,9 @@ def _func_custom_target_impl(self, node, args, kwargs):
                 mlog.warning('''Custom target input \'%s\' can\'t be converted to File object(s).
 This will become a hard error in the future.''' % kwargs['input'], location=self.current_node)
         kwargs['env'] = self.unpack_env_kwarg(kwargs)
+        if 'command' in kwargs and isinstance(kwargs['command'], list) and kwargs['command']:
+            if isinstance(kwargs['command'][0], str):
```

Comment:
Also if it's a mesonlib.File.

Reply:
I tried that and it does not work, because a File can point to a thing that will only be generated during build, so looking it up will fail.

Reply:
Hmm, I thought File was only for configure time generated files. OOC which API returns File for a build time file?

Reply:
I don't remember offhand, but if you change that and run project tests, you'll find out fairly quickly.

Reply:
Hmm, I tried with
```patch
diff --git a/mesonbuild/interpreter.py b/mesonbuild/interpreter.py
index 6a51f3d99..23a26896f 100644
--- a/mesonbuild/interpreter.py
+++ b/mesonbuild/interpreter.py
@@ -4013,6 +4013,10 @@ external dependencies (including libraries) must go to "dependencies".''')
                 mlog.warning('''Custom target input \'%s\' can\'t be converted to File object(s).
 This will become a hard error in the future.''' % kwargs['input'], location=self.current_node)
         kwargs['env'] = self.unpack_env_kwarg(kwargs)
+        command = mesonlib.extract_as_list(kwargs, 'command')
+        if command and isinstance(command[0], (str, mesonlib.File)):
+            command[0] = self.find_program_impl(command[0])
+            kwargs['command'] = command
         tg = CustomTargetHolder(build.CustomTarget(name, self.subdir, self.subproject, kwargs, backend=self.backend), self)
         self.add_target(name, tg.held_object)
         return tg
```

And project tests passed locally. Did you see it failing on the CI? Maybe there is another root cause?